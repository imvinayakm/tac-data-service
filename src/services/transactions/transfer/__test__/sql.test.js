const sql = require('../sql');

const filterValues = {
  recipient: 'recipient',
  assetId: 'someAssetID',
};

test('Sql search supports type-specific filters', () => {
  expect(sql.search(filterValues)).toMatchSnapshot();
});
