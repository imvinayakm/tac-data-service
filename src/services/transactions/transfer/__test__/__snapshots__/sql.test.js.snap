// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sql search supports type-specific filters 1`] = `"select \\"t\\".\\"height\\" as \\"height\\", \\"t\\".\\"tx_type\\" as \\"tx_type\\", \\"t\\".\\"id\\" as \\"id\\", \\"t\\".\\"time_stamp\\" as \\"time_stamp\\", \\"t\\".\\"signature\\" as \\"signature\\", \\"t\\".\\"proofs\\" as \\"proofs\\", \\"t\\".\\"tx_version\\" as \\"tx_version\\", t.fee * 10^(-fd.fee_asset_decimals) as \\"fee\\", \\"t\\".\\"sender\\" as \\"sender\\", \\"t\\".\\"sender_public_key\\" as \\"sender_public_key\\", \\"t\\".\\"asset_id\\" as \\"asset_id\\", t.amount * 10^(-ad.amount_asset_decimals) as \\"amount\\", \\"t\\".\\"recipient\\" as \\"recipient\\", \\"t\\".\\"fee_asset\\" as \\"fee_asset\\", \\"t\\".\\"attachment\\" as \\"attachment\\" from (select \\"height\\", \\"tx_type\\", \\"id\\", \\"time_stamp\\", \\"signature\\", \\"proofs\\", \\"tx_version\\", \\"fee\\", \\"sender\\", \\"sender_public_key\\", \\"asset_id\\", \\"amount\\", \\"recipient\\", \\"fee_asset\\", \\"attachment\\" from \\"txs_4\\" as \\"t\\" where \\"asset_id\\" = 'someAssetID' and \\"recipient\\" = 'recipient') as \\"t\\" inner join (select \\"asset_decimals\\".\\"decimals\\" as \\"amount_asset_decimals\\", \\"asset_decimals\\".\\"asset_id\\" as \\"asset_id\\" from \\"asset_decimals\\") as \\"ad\\" on \\"t\\".\\"asset_id\\" = \\"ad\\".\\"asset_id\\" inner join (select \\"asset_decimals\\".\\"decimals\\" as \\"fee_asset_decimals\\", \\"asset_decimals\\".\\"asset_id\\" as \\"asset_id\\" from \\"asset_decimals\\") as \\"fd\\" on \\"t\\".\\"fee_asset\\" = \\"fd\\".\\"asset_id\\" order by \\"time_stamp\\" asc, \\"id\\" asc"`;
